compiler/resolution/lowerIterators.cpp

```diff
@@ -1562,28 +1568,39 @@ static void processShadowVariables(ForLoop* forLoop, SymbolMap *map) {
 
         VarSymbol* capturedSvar = new VarSymbol(astr("cap_", svar->name), svar->type);
         forLoop->insertBefore(new DefExpr(capturedSvar));
-        forLoop->insertBefore(new CallExpr(PRIM_MOVE, capturedSvar, svar->outerVarSE->symbol()));
```

Comment:
Not related to this change, but I think you changed the name of the primitive in question. The comment above may need to be updated.

Reply:
I am also curious at a higher-level whether the comment above still explains what's going on here (I don't think so?). If it does, could you explain a bit more? Especially, I am not understanding the need for two different copies.

Reply:
For the `in` intent'd variable we want all tasks to get a private copy of the variable with the value initialized to whatever it was set to just before executing the loop.

I guess I've been calling task-private version a "task independent shadow variable capture". But now that I think about that I think this is a misnomer, so maybe we should rename the primitive to something like `PRIM_TASK_PRIVATE_SVAR_CAPTURE` instead.

So anyway, back to your question about why two different versions. You're asking why have:

```
var capX = x;
var taskCapX = PRIM_TASK_PRIVATE_SVAR_CAPTURE(capX);
```

instead of just, more directly:

```
var taskCapX = PRIM_TASK_PRIVATE_SVAR_CAPTURE(x);
```

I think my concern was if you have multiple tasks spawning at potentially  different times, there's a possibility of a race condition where something could modify `x` between different tasks starting.

Reply:
https://github.com/chapel-lang/chapel/pull/24451/commits/531de3672cac338323ddca789bb15004ae92a64b